contracts/storage/ProfileStorage.sol

```diff
@@ -2,135 +2,214 @@
 
 pragma solidity ^0.8.0;
 
-import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
-import {Hub} from '../Hub.sol';
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { ERC734 } from "../interface/ERC734.sol";
+import { HashingHub } from "../HashingHub.sol";
+import { Hub } from "../Hub.sol";
+
 
 contract ProfileStorage {
-    event AskUpdated(bytes nodeId, uint256 ask);
-    event StakeUpdated(bytes nodeId, uint256 stake);
+    event AskUpdated(bytes nodeId, uint96 ask);
+    event StakeUpdated(bytes nodeId, uint96 stake);
+    event RewardUpdated(bytes nodeId, uint96 reward);
 
     Hub public hub;
-    
+
+    struct ProfileDefinition{
+        uint96 ask;
+        uint96 stake;
+        uint96 reward;
+        uint96 stakeReserved;
+        uint96 withdrawalAmount;
+        bool withdrawalPending;
+        uint256 withdrawalTimestamp;
+        bytes nodeId;
+        mapping(uint8 => bytes32) hashedNodeIds;
```

Comment:
I would change this to "nodeAddresses", as they will effectively be different addresses in different key spaces